### PR TITLE
Enables Syndicate Paparazzi

### PR DESCRIPTION
**Pull request by SamHPurp**
:cl: Purpose2
add: Can now take photos of the Blueprints to count as a completed objective.
/ :cl:

### CODE REVIEW

**Comment by Fruerlund** [comment]
Thage her beat 👌

**Comment by Fruerlund** [comment]
Thats very neat, damn autocorrect

File: code/game/gamemodes/objective.dm [hunk 1]
```diff
@@ -384,8 +384,18 @@ var/list/potential_theft_objectives = subtypesof(/datum/theft_objective) \
...
+
+	var/list/all_items = owner.current.GetAllContents()
+
+	for(var/obj/I in all_items)
+		if(istype(I, steal_target))
+			return 1
+		if (steal_target.check_special_completion(I))
+			return 1
+
+	return 0
```
Comment (Fox-McCloud) [inline code/game/gamemodes/objective.dm]: I don't think this needs to be changed; as a matter of fact, I think this may break things.

**Comment by jughu** [comment]
Make it require to be a 1x1 photo

File: code/game/gamemodes/steal_items.dm [hunk 2]
```diff
@@ -71,6 +71,15 @@ datum/theft_objective/ai/check_special_completion(var/obj/item/device/aicard/C)
...
 	protected_jobs = list("Chief Engineer")
 
+/datum/objective_item/steal/blueprints/check_special_completion(obj/item/I)
+	if(istype(I, /obj/item/areaeditor/blueprints))
+		return 1
+	if(istype(I, /obj/item/weapon/photo))
+		var/obj/item/weapon/photo/P = I
+		if(P.blueprints)
+			return 1
+	return 0
```
Comment (Fox-McCloud) [inline code/game/gamemodes/steal_items.dm]: I *think* just

```
/datum/objective_item/steal/blueprints/check_special_completion(obj/item/I)
	if(..())
		if(istype(I, /obj/item/areaeditor/blueprints))
			return 1
		if(istype(I, /obj/item/weapon/photo))
			var/obj/item/weapon/photo/P = I
			if(P.blueprints)
				return 1
	return 0
```

Will cover this

**Comment by jughu** [comment]
I mean its not like syndiecate want a photo where you can barely see the blueprints

File: code/game/gamemodes/objective.dm (hunk 1 above)
Comment (SamHPurp) [inline code/game/gamemodes/objective.dm]: This works @Fox-McCloud 
image

File: code/game/gamemodes/objective.dm [hunk 3]
```diff
@@ -384,8 +384,18 @@ var/list/potential_theft_objectives = subtypesof(/datum/theft_objective) \
...
 /datum/objective/steal/check_completion()
-	if(!steal_target) return 1 // Free Objective
-	return steal_target.check_completion(owner)
+	if(!steal_target)
+		return 1 // Free Objective
+
+	var/list/all_items = owner.current.GetAllContents()
+
+	for(var/obj/I in all_items)
+		if(istype(I, steal_target))
```
Comment (RemieRichards) [inline code/game/gamemodes/objective.dm]: this is a ```/datum/theft_objective```, so this needs to be ```if(istype(I, steal_target.typepath))```, now this is different to how /tg/ does it, and I've only just glanced at your code, but @Fox-McCloud confirm?

File: code/game/gamemodes/objective.dm [hunk 4]
```diff
@@ -384,8 +384,18 @@ var/list/potential_theft_objectives = subtypesof(/datum/theft_objective) \
...
+			return 1
```
Comment (RemieRichards) [inline code/game/gamemodes/objective.dm]: this should be ```return steal_target.check_special_completion(I)```
(then removing the second if below)

File: code/game/gamemodes/objective.dm (hunk 3 above)
Comment (SamHPurp) [inline code/game/gamemodes/objective.dm]: @RemieRichards Thats the conclusion we reached as well!

Comment (RemieRichards) [inline code/game/gamemodes/objective.dm]: Ah splendid, good to know my brain still works before lunch.

Comment (SamHPurp) [inline code/game/gamemodes/objective.dm]: @RemieRichards embarressingly it works for the things I was testing... AIs, slime cores, blueprints.... but breaks and auto-greentexts everything else. Red cheeks.

**Comment by Fox-McCloud** [comment]
Hai, @RemieRichards. Funny you should post what you did; about 10 minutes before you posted, I told him pretty much what you did, then stated something along the lines of "give yourself the steal the HoS's energy gun objective; I bet it greentexts if you have a single object on you"

**Comment by RemieRichards** [comment]
Heh.

**Comment by SamHPurp** [comment]
@Fox-McCloud So I fixed this. 

Objectives now have an alternate item list, so if this does need to expand to other things in the future, this is easy to do.

- Succeeds having blueprints and the photo of the blueprints
- Fails not having the items for other objectives (this time, lol).
- Special tests all succeeded, IE, empty slime cores, empty AI cards, unfull plasma tanks

File: code/game/gamemodes/steal_items.dm [hunk 5]
```diff
@@ -6,10 +6,11 @@
...
-	var/name=""
+	var/name = ""
 	var/typepath=/atom
-	var/list/protected_jobs=list()
-	var/flags=0
+	var/list/protected_jobs = list()
+	var/list/altitems = list()
+	var/flags = 0
 
 /datum/theft_objective/proc/check_completion(var/datum/mind/owner)
```
Comment (Fox-McCloud) [inline code/game/gamemodes/steal_items.dm]: I don't think this is even used anymore.

**Comment by Xantholne** [comment]
I could've sworn this was already a thing a few years back.

**Comment by SamHPurp** [comment]
@Xantholne not on Paradise, but on TG, and maybe even Bay.

File: code/game/gamemodes/steal_items.dm (hunk 5 above)
Comment (Fox-McCloud) [inline code/game/gamemodes/steal_items.dm]: This comment still remains @SamHPurp

Comment (SamHPurp) [inline code/game/gamemodes/steal_items.dm]: @Fox-McCloud will review in the morning.

Comment (SamHPurp) [inline code/game/gamemodes/steal_items.dm]: @Fox-McCloud this is still used, for checking credits for a 'money theft' objective... which I've never seen actually utilised.

Want me to leave it? Or rip out the whole money-theft thing.

Comment (SamHPurp) [inline code/game/gamemodes/steal_items.dm]: Actually this is also used to determine the fullness of a Plasma tank for the Plasma objective.

Might be worth leaving it, unless you have a better suggestion?

Comment (taukausanake) [inline code/game/gamemodes/steal_items.dm]: Wait, you can steal money? Why is that not an objective? That would be great to steal credits from other peoples accounts

Comment (Fox-McCloud) [inline code/game/gamemodes/steal_items.dm]: REEEeeeeeEEEeeEEEEe.

Objective code. This can just be refactored later, I guess.

**Pull request merged**